src/components/narrativeEditor/examineNarrative.js

```diff
@@ -0,0 +1,275 @@
+   * As the debug narrative page is still a prototype I attempted to avoid modifying too many existing auspice
+   * functions, but in the future we should refactor `loadNarrative` so that we can reuse
+   * the core functionality.
+   * We explicity make a CLEAN_START each time so we don't have to worry about
+   * previous redux state (e.g. from a previous narrative in the editor)
+   *
+   * There is one (subtle) difference between loading a normal narrative and
+   * this approach with regards to the initial dataset if the frontmatter dataset
+   * isn't valid and the 1st slide's dataset is (and vice versa). As we're debugging
+   * I want us to flag this as an error rather than being clever about it.
+   * @param {int} n
+   */
+  const showNarrative = async (n=0) => {
+    const datasetNames = summary[n].associatedDatasetNames;
+
+    let cleanStartAction;
+    try {
+      cleanStartAction = {
+        type: CLEAN_START,
+        pathnameShouldBe: undefined,
+        preserveCache: true, // specific for the narrative debugger
+        ...createStateFromQueryOrJSONs({
+          json: await narrative.datasets[datasetNames[0]].main,
+          secondTreeDataset: datasetNames[1] ? await narrative.datasets[datasetNames[1]].main : undefined,
+          query: n ? {n} : {}, // query is things like n=3 to load a specific page
+          narrativeBlocks: narrative.blocks,
+          mainTreeName: narrative.datasets[narrative.initialNames[0]].pathname,
+          secondTreeName: datasetNames[1] || null,
+          dispatch
+        })
+      };
+    } catch (err) {
+      // todo
+      console.error(`Failed to load the narrative on slide ${n}`, err);
+      return;
+    }
+    dispatch(cleanStartAction);
+    setDisplayNarrative(true);
+
+    /* the CLEAN_START action _does not_ consider sidecars, so we do this manually... */
+    narrative.datasets[datasetNames[0]].loadSidecars(dispatch);
+  };
+
+  return (
+    <div ref={el}>
+      <ExplainSlides/>
+      <NarrativeSummary summary={summary} datasetResponses={datasetResponses} showNarrative={showNarrative}/>
+      <Styles.ButtonContainer>
+        <Styles.Button onClick={() => showNarrative()}>
+          View Narrative
+        </Styles.Button>
+      </Styles.ButtonContainer>
+      <div style={{minHeight: "50px"}}/>
+    </div>
+  );
+};
+
+/**
+ * Our narrative parsing functions don't return the underlying markdown, but
+ * because we control the HTML output we can make certain assumptions about
+ * where the title is. This function should be considered fit for purpose only
+ * for the MVP -- we should instead modify `parseMarkdownNarrativeFile`
+ * to return this information
+ * @param {string} html
+ * @returns {string} the title
+ */
+function extractTitleFromHTML(html) {
+  const reFrontmatterSlide = /^<h2[^>]+>(.+?)<\/h2>/;
+  const reNormalSlide = /^<h1[^>]+>(.+?)<\/h1>/;
+  try {
+    if (html.match(reFrontmatterSlide)) {
+      return html.match(reFrontmatterSlide)[1];
+    }
```

Comment:
It's not overly hard but I wanted to avoid changing any of the actual narratives machinery in auspice for this MVP to minimise any unintended side-effects. I just about succeeded. For this particular function I've moved to a `DOMParser` in https://github.com/nextstrain/auspice/compare/3ddbd1616ce697044c3c690eebc054226681335f..1adfb5c737d3393ddf22081c94cda3de8b4f7913